app/components/build/AvatarWithUnknownEmailPrompt.js

```diff
@@ -0,0 +1,321 @@
+    relay: React.PropTypes.object
+  };
+
+  state = {
+    isAddingEmail: false,
+    isDismissing: false,
+    isSendingVerification: false,
+    hasSentSomething: false,
+    hasBeenDismissed: false
+  };
+
+  getUserEmail(email) {
+    const userEmails = this.props.viewer.emails.edges;
+
+    const userEmail = userEmails.find(
+      ({ node: { address: userEmail } }) => (
+        userEmail.toLowerCase() === email.toLowerCase()
+      )
+    );
+
+    if (userEmail) {
+      return userEmail.node;
+    }
+  }
+
+  isCurrentUsersValidatedEmail(email) {
+    const userEmail = this.getUserEmail(email);
+    return userEmail && userEmail.verified;
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return shallowCompare(this, nextProps, nextState);
+  }
+
+  componentDidMount() {
+    const { createdBy } = this.props.build;
+
+    if (createdBy && createdBy.email && !this.isCurrentUsersValidatedEmail(createdBy.email)) {
+      this.props.relay.setVariables(
+        {
+          isTryingToPrompt: true,
+          emailForPrompt: createdBy.email.toLowerCase()
+        },
+        (readyState) => {
+          if (readyState.done) {
+            setTimeout(() => {
+              const { dismissedAt } = this.props.viewer.notice;
+
+              if (this._dropdown && !dismissedAt) {
+                this._dropdown.setShowing(true);
+              }
+            }, 0);
+          }
+        }
+      );
+    }
+  }
+
+  handleMutationFailure = (transaction) => {
+    this.setState({ isAddingEmail: false, isDismissing: false, isSendingVerification: false });
+
+    FlashesStore.flash(FlashesStore.ERROR, transaction.getError());
+  };
+
+  handleDismissClick = () => {
+    this.setState({ isDismissing: true });
+
+    const mutation = new NoticeDismissMutation({ viewer: this.props.viewer, notice: this.props.viewer.notice });
+
+    Relay.Store.commitUpdate(mutation, { onSuccess: this.handleDismissedSucess, onFailure: this.handleMutationFailure });
+
+    this._dropdown.setShowing(false);
+  };
+
+  handleAddEmailClick = () => {
+    this.setState({ isAddingEmail: true });
+
+    const mutation = new EmailCreateMutation({ address: this.props.build.createdBy.email, viewer: this.props.viewer });
+
+    Relay.Store.commitUpdate(mutation, { onSuccess: this.handleEmailAddedSuccess, onFailure: this.handleMutationFailure });
+  };
+
+  handleResendVerificationClick = () => {
+    this.setState({ isSendingVerification: true });
+
+    const mutation = new EmailResendVerificationMutation({ email: this.getUserEmail(this.props.build.createdBy.email) });
+
+    Relay.Store.commitUpdate(mutation, { onSuccess: this.handleVerificationResentSuccess, onFailure: this.handleMutationFailure });
+  };
+
+  handleDismissedSucess = () => {
+    this.setState({ isDismissing: false });
+  };
+
+  handleEmailAddedSuccess = () => {
+    this.setState({ isAddingEmail: false, hasSentSomething: true });
+  };
+
+  handleVerificationResentSuccess = () => {
```

Comment:
And this - I don't think we do verification sending from the dialog anymore, so it's associated bits and pieces can be ditched!